### PR TITLE
[Microsoft Teams] Fix URL RegEx

### PR DESCRIPTION
**Pull request by MichaelYochpaz**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-7237)
## Description
Fix URL RegEx which previously didn't capture URL parameters (ex: `?param=1`) to capture any string that starts with http/https.

## Screenshots
Before and after:
Screenshot 2023-07-02 at 13 54 17

### CODE REVIEW

**Comment by MichaelYochpaz** [comment]
@bziser Many of the changes are due to ruff autofixes. You can see a diff without them (only with fix-related changes) here:
https://github.com/demisto/content/pull/27846/files/b22390a9c9c202c53e4ad5e9e629e3149cf71d61

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 1]
```diff
@@ -288,16 +288,16 @@ def is_investigation_mirrored(investigation_id: str, mirrored_channels: list) ->
...
+    :return: Formatted message with hyperlinks.
     """
     url_header = url_header or EXTERNAL_FORM_URL_DEFAULT_HEADER
     formatted_message: str = message
-    # URLify markdown hyperlinks
-    urls = re.findall(URL_REGEX, message)
-    for url in urls:
+
+    for url_match in re.finditer(urlRegex, message):
+        url: str = url_match.group()
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:300]: why needed `:str`?

**Review by bziser — COMMENTED** [review]
Looks good

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py (hunk 1 above)
Comment (MichaelYochpaz) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:300]: The `group()` method returns `AnyStr` and not `str`, while `replace` receives `str`, which seems to confuse IDEs (both VSCode and PyCharm complain)
Not sure if it fails mypy checks though.

Screenshot 2023-07-02 at 14 15 08

**Review by bziser — APPROVED** [review]

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 2]
```diff
@@ -222,15 +222,15 @@ def process_incident_create_message(demisto_user: dict, message: str, request_bo
     json_pattern: str = r'(?<=json=).*'
     name_pattern: str = r'(?<=name=).*'
     type_pattern: str = r'(?<=type=).*'
-    json_match: Optional[Match[str]] = re.search(json_pattern, message)
-    created_incident: Union[dict, list] = []
-    data: str = str()
+    json_match: Match[str] | None = re.search(json_pattern, message)
+    created_incident: dict | list = []
+    data: str = ''
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:227]:
```suggestion
    data = ''
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 3]
```diff
@@ -310,7 +310,7 @@ def get_team_member(integration_context: dict, team_member_id: str) -> dict:
     :param team_member_id: Team member ID to search for
     :return: Found team member object
     """
-    team_member: dict = dict()
+    team_member: dict = {}
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:313]:
```suggestion
    team_member = {}
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 4]
```diff
@@ -373,7 +373,7 @@ def process_tasks_list(data_by_line: list) -> dict:
     :param data_by_line: List of tasks to process
     :return: Adaptive card of assigned tasks
     """
-    body: list = list()
+    body: list = []
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:376]:
```suggestion
    body = []
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 5]
```diff
@@ -406,7 +406,7 @@ def process_incidents_list(data_by_line: list) -> dict:
     :param data_by_line: List of incidents to process
     :return: Adaptive card of assigned incidents
     """
-    body: list = list()
+    body: list = []
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:409]:
```suggestion
    body = []
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 6]
```diff
@@ -477,7 +477,7 @@ def process_ask_user(message: str) -> dict:
             'text': text
         }
     ]
-    actions: list = list()
+    actions: list = []
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:480]:
```suggestion
    actions = []
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 7]
```diff
@@ -757,7 +757,7 @@ def validate_auth_header(headers: dict) -> bool:
 
     unverified_headers: dict = jwt.get_unverified_header(jwt_token)
     key_id: str = unverified_headers.get('kid', '')
-    key_object: dict = dict()
+    key_object: dict = {}
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:760]:
```suggestion
    key_object = {}
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 8]
```diff
@@ -1957,8 +1957,8 @@ def send_message():
         raise ValueError('Provide either message or adaptive to send, not both.')
 
     integration_context: dict = get_integration_context()
-    channel_id: str = str()
-    personal_conversation_id: str = str()
+    channel_id: str = ''
+    personal_conversation_id: str = ''
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1961]:
```suggestion
    channel_id = ''
    personal_conversation_id = ''
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 9]
```diff
@@ -2012,7 +2012,7 @@ def get_channel_id_for_send_notification(channel_name: str, message_type: str):
     """
     team_name: str = demisto.args().get('team', '') or demisto.params().get('team', '')
     team_aad_id: str = get_team_aad_id(team_name)
-    investigation_id: str = str()
+    investigation_id: str = ''
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:2015]:
```suggestion
    investigation_id = ''
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 10]
```diff
@@ -2043,9 +2043,9 @@ def mirror_investigation():
     if not team_name:
         team_name = demisto.params().get('team', '')
     team_aad_id: str = get_team_aad_id(team_name)
-    mirrored_channels: list = list()
+    mirrored_channels: list = []
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:2046]:
```suggestion
    mirrored_channels = []
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 11]
```diff
@@ -2043,9 +2043,9 @@ def mirror_investigation():
...
     teams: list = json.loads(integration_context.get('teams', '[]'))
-    team: dict = dict()
+    team: dict = {}
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:2048]:
```suggestion
    team = {}
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 12]
```diff
@@ -2071,7 +2071,7 @@ def mirror_investigation():
         service_url: str = integration_context.get('service_url', '')
         server_links: dict = demisto.demistoUrls()
         server_link: str = server_links.get('server', '')
-        warroom_link: str = f'{server_link}#/WarRoom/{investigation_id}'
+        warroom_link: str = f'{server_link}/#/WarRoom/{investigation_id}'
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:2074]:
```suggestion
        warroom_link = f'{server_link}/#/WarRoom/{investigation_id}'
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 13]
```diff
@@ -2248,9 +2248,9 @@ def direct_message_handler(integration_context: dict, request_body: dict, conver
     user_email: str = team_member.get('user_email', '')
     demisto_user = demisto.findUser(email=user_email) if user_email else demisto.findUser(username=username)
 
-    formatted_message: str = str()
+    formatted_message: str = ''
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:2251]:
```suggestion
    formatted_message = ''
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 14]
```diff
@@ -2248,9 +2248,9 @@ def direct_message_handler(integration_context: dict, request_body: dict, conver
...
 
-    attachment: dict = dict()
+    attachment: dict = {}
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:2253]:
```suggestion
    attachment = {}
```

**Review by bziser — COMMENTED** [review]
please check if the url must have `.com`

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 15]
```diff
@@ -239,14 +239,14 @@ def process_incident_create_message(demisto_user: dict, message: str, request_bo
...
                 data = 'Failed creating incidents.'
     else:
-        name_match: Optional[Match[str]] = re.search(name_pattern, message)
+        name_match: Match[str] | None = re.search(name_pattern, message)
         if not name_match:
             data = 'Please specify arguments in the following manner: name=<name> type=[type] or json=<json>.'
         else:
             incident_name: str = re.sub('type=.*', '', name_match.group()).strip()
-            incident_type: str = str()
+            incident_type: str = ''
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:247]:
```suggestion
            incident_type = ''
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 16]
```diff
@@ -1806,7 +1806,7 @@ def close_channel():
     channel_name: str = demisto.args().get('channel', '')
     investigation: dict = demisto.investigation()
     investigation_id: str = investigation.get('id', '')
-    channel_id: str = str()
+    channel_id: str = ''
```
Comment (bziser) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1809]:
```suggestion
    channel_id = ''
```

**Review by bziser — COMMENTED** [review]
great

**Review by bziser — APPROVED** [review]

**Pull request merged**